src/backup.rs

```diff
@@ -26,22 +29,52 @@ struct Backup {
 
 
 impl BackupOptions {
-    pub fn backup(&self, archive_path: &Path, source: &Path, report: &Report) -> Result<()> {
+    pub fn backup(&self, archive_path: &Path, source: &Path, report: &Report, excludes: Option<Vec<&str>>) -> Result<()> {
         let archive = try!(Archive::open(archive_path, &report));
         let band = try!(archive.create_band(&report));
+        let excludes = BackupOptions::get_excludes(excludes)?;
+
         let mut backup = Backup {
             block_dir: band.block_dir(),
             index_builder: band.index_builder(),
             report: report.clone(),
         };
+
         let source_iter = try!(sources::iter(source, report));
         for entry in source_iter {
-            try!(backup.store_one_source_entry(&try!(entry)));
+            let entry = entry?.clone();
```

Comment:
Let's do the exclusions inside of `sources` rather than on the output, or at least make it a filter on the iter.

Actually, doing it in Sources will let us avoid traversing ignored directories.